### ISSUE TITLE
getMyLocation not working

### ISSUE THREAD

**Issue by cedrickbae** [ionic-team/ionic-native-google-maps#7]
getMyLocation not working, I tried to alert the result and the error but nothing happens.

This is my ionic cordova plugin

```
cordova-android-play-services-gradle-release 1.3.0 "cordova-android-play-services-gradle-release"
cordova-plugin-device 2.0.1 "Device"
cordova-plugin-googlemaps 2.2.9 "cordova-plugin-googlemaps"
```

and here is my function
```
get_location(){
this.map.getMyLocation().then((pos) => {
      alert(JSON.stringify(pos));
},err => alert(JSON.stringify(err));
}
```

**Comment by wf9a5m75** [comment]
https://github.com/mapsplugin/ionic-googlemaps-quickdemo/blob/master/src/pages/home/home.ts#L56-L82

**Comment by wf9a5m75** [comment]
Please share your project files on github.

**Comment by battika** [comment]
Are you using Genymotion for testing this? I have mixed results when using Genymotion it works sometimes but it fails most of the time. Try it on a real device and see if you have issues.

**Comment by wf9a5m75** [comment]
@battika Please try the multiple_maps branch. The getMyLocation() should work on Genymotion. (I already confirmed)

**Comment by cedrickbae** [comment]
I'm using a real device for testing this. 

`this.map.setMyLocationEnabled(true);`
works and shows on map but getMyLocation() does not.

**Comment by wf9a5m75** [comment]
As I said `Please share your project files on github.`

**Comment by battika** [comment]
I have a look too using my test project I wanted to update it anyways. I shall let you know what I found.

**Comment by battika** [comment]
Oh, just realized you are calling `this.map.getMyLocation()` to get your location. I tried and it did not work for me either. 
However, if you use the recently introduced LocationService it works flawlessly like this:
```
    LocationService.getMyLocation({enableHighAccuracy: true}).then(
        (result) => {
          console.log(JSON.stringify(result));
        },
        (err) => {
          console.error(JSON.stringify(err));
        }
    ).catch((e) => {
      console.error(JSON.stringify(e));
    });
```
You need to import LocationService in your import section.

Just updated my Google Maps Ionic example, please find the file with the example here:

https://github.com/battika/ionic-gmaps-test/blob/master/src/pages/home/home.ts

**Comment by cedrickbae** [comment]
Thanks @battika! It now works!

**Comment by battika** [comment]
I am glad it helped. Not sure why the one you tried failed.

**Comment by cedrickbae** [comment]
Yeah, I wonder too, initially this.map.getMyLocation() works well when I install it in ionic doc page, but after I install it in this repo, the error occurred.
I now use LocationService.getMyLocation() and it works fine. Again thanks!

**Comment by Hanzofm** [comment]
Using static `LocationService`, if use `{enableHighAccuracy: true}` param on android 8.1 the promise not returns values and also not return any error.

Removes the param solves temporally the problem. `{enableHighAccuracy: true}` work perfectly on Android 7.0

**Comment by mateuswetah** [comment]
@Hanzofm did you manage to solve this? If not I am going to to open a new issue, I'm facing exactly this: either _this.map.getMyLocation_ or _LocationService.getMyLocation_ only work when I don't provide the `{enableHighAccuracy: true}`. Otherwise its infinite loading...

**Comment by mateuswetah** [comment]
BTW, I was able to run the exact same code using that other plugin, the Geolocation one. So:
`this.geoLocation.getCurrentPosition({ enableHighAccuracy: true })` works, but `this.map.getMyLocation({ enableHighAccuracy: true })` does not :(

**Comment by wf9a5m75** [comment]
Pull request is welcome :)

**Comment by mateuswetah** [comment]
@wf9a5m75 I acknowledge the complexity of keeping a project like this but I really have no expertise to do some direct contribution right now. If it helps in any way, I can comment some details:
- Before moving into the Geolocation plugin, I tried the famous `multiple_maps` branch;
- I also make sure that I was running my code after the `this.platform.ready()` promise finishes;
- I did tried both `LocationService.getMyLocation()` and `this.map.getMyLocation()`. 

While none of this worked, I can comment one detail on the Geolocation implementation that calls my attention: it requires to be listed in the components `providers` list. I am not sure how Angular internals work so I'm not sure if this could affect any of it.

Thanks anyways for the rest of the work on the plugin, it still does a hell of a job!
